service/smithyprototype/lexruntimeservice/deserialize.go

```diff
@@ -0,0 +1,1444 @@
+package lexruntimeservice
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/sdkio"
+	"github.com/aws/aws-sdk-go-v2/service/smithyprototype/lexruntimeservice/types"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyHTTP "github.com/awslabs/smithy-go/transport/http"
+)
+
+type awsRestJson1_deserializeOpPutSession struct{}
+
+// ID is the middleware identifier
+func (p awsRestJson1_deserializeOpPutSession) ID() string {
+	return "awsRestJson1_deserializeOpPutSession"
+}
+
+func (p awsRestJson1_deserializeOpPutSession) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
+	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	out, metadata, err = next.HandleDeserialize(ctx, in)
+	if err != nil {
+		return out, metadata, err
+	}
+	response, ok := out.RawResponse.(*smithyHTTP.Response)
+	if !ok {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("unknown transport type %T", out.RawResponse),
+		}
+	}
+
+	output, ok := out.Result.(*PutSessionOutput)
+	if !ok {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("unknown output result type %T", out.Result),
+		}
+	}
+
+	if err := awsRestJson1_deserializeRestPutSessionOutput(output, response); err != nil {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("failed to decode response with invalid REST bindings, %w", err),
+		}
+	}
+
+	// body of type blob
+	if err := awsRestJson1_deserializeJsonPutSessionOutput(output, response.Body); err != nil {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("failed to decode response body with invalid JSON, %w", err),
+		}
+	}
```

Comment:
Nit outside of the scope for this PR, but should would be handy to add a TODO before the `awsRestJson1_deserializeRestPutSessionOutput` for error deserializing.